internal/controllers/utils/utils.go

```diff
@@ -163,8 +166,88 @@ func GetDeploymentVolumeMounts(serverName string) []corev1.VolumeMount {
+		return fmt.Sprintf("--backend-token=%s", orano2ims.Spec.BackendToken)
+	}
+
+	return fmt.Sprintf("--backend-token-file=%s", defaultBackendTokenFile)
+}
+
+// getACMNamespace will determine the ACM namespace from the multiclusterengine object.
+//
+// multiclusterengine object sample:
+//   apiVersion: multicluster.openshift.io/v1
+//   kind: MultiClusterEngine
+//   metadata:
+//     labels:
+//       installer.name: multiclusterhub
+//       installer.namespace: open-cluster-management
+func getACMNamespace(ctx context.Context, c client.Client) (string, error) {
+	// Get the multiclusterengine object.
+	multiClusterEngine := &unstructured.Unstructured{}
+	multiClusterEngine.SetGroupVersionKind(schema.GroupVersionKind{
+		Group: "multicluster.openshift.io",
+		Kind: "MultiClusterEngine",
+		Version: "v1",
+	})
+	err := c.Get(ctx, client.ObjectKey{
+		Name:      "multiclusterengine",
+	}, multiClusterEngine)
+
+	if err != nil {
+		oranUtilsLog.Info("[getACMNamespace] multiclusterengine object not found")
+		return "", fmt.Errorf("multiclusterengine object not found")
+	}
+
+	// Get the ACM namespace by looking at the installer.namespace label.
+	multiClusterEngineMetadata := multiClusterEngine.Object["metadata"].(map[string]interface{})
+	multiClusterEngineLabels, labelsOk := multiClusterEngineMetadata["labels"]
+
+	if labelsOk {
+		acmNamespace, acmNamespaceOk := multiClusterEngineLabels.(map[string]interface{})["installer.namespace"]
+
+		if !acmNamespaceOk {
+			return "", fmt.Errorf("multiclusterengine labels does not contain the installer.namespace key")
+		}
+		return acmNamespace.(string), nil
+	}
+
+	return "", fmt.Errorf("multiclusterengine object does not have expected labels")
+}
+
+// buildSearchAPI will dynamically obtain the search API.
+func buildSearchAPI(ctx context.Context, c client.Client, orano2ims *oranv1alpha1.ORANO2IMS) (string, error) {
+	// Find the ACM namespace.
+	acmNamespace, err := getACMNamespace(ctx, c)
+	if err != nil {
+		return "", err
+	}
+
+	// Split the Ingress to obtain the domain for the Search API.
+	// searchAPIBackendURL example: https://search-api-open-cluster-management.apps.lab.karmalabs.corp
+	// IngressHost example:         o2ims.apps.lab.karmalabs.corp
+	// Note: The domain could also be obtained from the spec.host of the search-api route in the
+	// ACM namespace.
+	ingressSplit := strings.Split(orano2ims.Spec.IngressHost, ".apps")
+	if len(ingressSplit) != 2 {
+		return "", fmt.Errorf(`the searchAPIBackendURL could not be obtained from the IngressHost.
+			Directly specify the searchAPIBackendURL in the ORANO2IMS CR or update the IngressHost`)
+	}
+	domain := ".apps" + ingressSplit[len(ingressSplit)-1]
+
+	// The searchAPI is obtained from the "search-api" string and the ACM namespace.
+	searchAPI := "https://" + "search-api-" + acmNamespace + domain
+
+	return searchAPI, nil
+}
+
+func BuildServerContainerArgs(ctx context.Context, c client.Client, orano2ims *oranv1alpha1.ORANO2IMS,
```

Comment:
The number of parameters of this function is growing, as usually happens, at some point we should consider making it a method of some type, so that we can have some of the parameters as fields of the type and avoid adding parameters. I just want to avoid things like this, which happen over time (from real code):

```go
clusterApi = NewManager(getDefaultConfig(), common.GetTestLog(), db, commontesting.GetDummyNotificationStream(ctrl), nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, false)
```

Reply:
Agreed. Noted.